src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,166 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (tail == null && head == null) {
+            addFirst(value);
+        } else if (index == 0) {
+            Node<T> node = new Node<>(null, value, head);
+            head.prev = node;
+            head = node;
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            Node<T> node = new Node<>(prev, value, prev.next);
+            prev.next.prev = node;
+            prev.next = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        T result = node.value;
+        node.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, size);
+        Node<T> node = getNodeByIndex(index);
+        unlink(node);
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndexByValue(object);
+        if (index == -1) {
+            return false;
+        }
+        unlink(getNodeByIndex(index));
```

Comment:
there is no sense to do so much stuff here - you iterate first to find an index and then you iterate to find a node

why not using a single for loop here to find a desired node and then call `unlink`?

basically you can remove `getIndexByValue` and use the loop from that method here